src/workerd/api/actor.h

```diff
@@ -13,27 +13,82 @@
 #include <capnp/compat/http-over-capnp.h>
 #include <workerd/api/http.h>
 #include <workerd/jsg/jsg.h>
+#include <workerd/jsg/function.h>
 
 namespace workerd {
   template <typename T> class IoOwn;
 }
 
 namespace workerd::api {
 
+enum IsInHouse {
+  YES,
+  NO,
+};
+
+// This class is the client RPC stub for Durable Objects. It intercepts dynamic method calls, then
+// sends the method name and arguments to the remote Durable Object, returning the response it
+// receives.
+//
+// We don't want name interception on Fetcher itself -- at least not yet -- since not all Fetcher
+// implementations need it. Name interception is off by default; to turn it on, set the JsRpcStub
+// compatibility flag.
+class DurableObjectRpcStub : public Fetcher, public jsg::NamedIntercept {
+public:
+  DurableObjectRpcStub(
+      IoOwn<OutgoingFactory> outgoingFactory,
+      RequiresHostAndProtocol requiresHost,
+      IsInHouse inHouse,
+      CompatibilityFlags::Reader flags)
+    : Fetcher(kj::mv(outgoingFactory), requiresHost, (inHouse == YES)), flags(flags) {}
+
+  kj::Maybe<jsg::JsValue> getNamed(jsg::Lock& js, kj::StringPtr name) override {
+    if (flags.getJsRpcStub()) {
+      // Named intercept is enabled, this means we won't default to legacy behavior.
+      // TODO(soon): The server side PR will fill this function with the RPC call and other stuff,
+      //             this is just a temporary placeholder for testing.
+      // TODO(now): Do we actually need JSG_VISITABLE_LAMBDA? We are passing JS args so maybe?
+      return jsg::JsValue(js.wrapReturningFunction(js.v8Context(), JSG_VISITABLE_LAMBDA(
```

Comment:
In this case `JSG_VISITABLE_LAMBDA` is probably unnecessary. If/when this accepts any actual captures that could change.

Reply:
https://github.com/cloudflare/workerd/compare/f26b6b44b6f101d63f3d6883bb363a7dedbba110..1715a12c14db2e5001deb311bfd6062954be2db1